index.bs

```diff
@@ -177,11 +177,6 @@ dictionary DeviceOrientationEventInit : EventInit {
     double? gamma = null;
     boolean absolute = false;
 };
-
-enum PermissionState {
```

Comment:
In specification text an additional value `"default"` is also mentioned. These references should be cleaned up to use `"prompt"` instead.

Reply:
The "Permission model" section seems also redundant and should be removed.

Reply:
Oops, this dropped off my radar, I didn't notice the comments last year.

I made an attempt to refresh the patch now to use terms from https://w3c.github.io/permissions/, but it looks like there is no permission for this in https://w3c.github.io/permissions/#permission-registry?

I think one has to invoke https://w3c.github.io/permissions/#request-permission-to-use with a string, but there is no string like "deviceorientation" for this. What to do?

Reply:
I believe both Blink and WebKit use:

1. DeviceMotionEvent: Requires gyroscope & accelerometer
2. DeviceOrientationEvent: Requires gyroscope & accelerometer & magnetometer

@cdumez via https://bugs.webkit.org/show_bug.cgi?id=221399#c13

Reply:
Hmm, I don't think Gecko and Chrome require permission for these APIs, per my quick test. https://bugzilla.mozilla.org/show_bug.cgi?id=1536382 implies Safari was implementing this, but I have no Apple device, could you do a quick check?

Reply:
Chromium grants these permissions by default but provides user controls to disable them. We have not yet implemented the requestPermission() method. This is tracked by [issue 947112](https://bugs.chromium.org/p/chromium/issues/detail?id=947112) and is blocked on some settings UX questions.

I support using the mapping to Permission API names suggested above. In browsers which implement the Permissions API and/or the Generic Sensors API equivalents to these events the consistency should be good for developers. In browsers which do not I don't think it makes a difference.

Reply:
@ reillyeon
> In specification text an additional value `"default"` is also mentioned. These references should be cleaned up to use `"prompt"` instead.

I've also found that https://notifications.spec.whatwg.org/#permission-model also uses `"default"` instead of `"prompt"`.

How breaking the changing from `"default"` to `"prompt"`?

(I personally would love to see `"prompt"` instead of `"default"` in both specs.)

Reply:
> How breaking the changing from `"default"` to `"prompt"`?

In this case, I don't think it would be a breaking change because contrary to the Notifications API spec, "default" is not exposed as part of the API here, it's an internal state used in the spec.

Unfortunately, also contrary to the Notifications API spec there's no integration with the Permissions API here whatsoever (see #70). Implementing that integration would allow us to get rid of the "Permission model" section, some of the existing text and likely simplify the `requestPermission()` algorithms, but then we also go back to some of the comments above about what powerful features we'd requesting permission for (and how to avoid breaking WebKit depending on the values we use).

Reply:
Let's clean up the `"default"` value in the follow-up which integrates with the Permissions API. It doesn't look technically necessary in this change.